CoffeeChat/View/OnboardingSearchDropdownView.swift

```diff
@@ -0,0 +1,153 @@
+//
+//  OnboardingSearchDropdownView.swift
+//  CoffeeChat
+//
+//  Created by Lucy Xu on 2/13/20.
+//  Copyright © 2020 cuappdev. All rights reserved.
+//
+
+import UIKit
+
+
+protocol OnboardingSearchViewDelegate: class {
+    func bringSearchViewToFront(searchView: UIView, height: CGFloat, isSelect: Bool)
+    func sendSearchViewToBack(searchView: UIView)
+    func updateSearchViewHeight(searchView: UIView, height: CGFloat)
+    func updateSelectedFields(tag: Int, isSelected: Bool)
+}
+
+/// Custom onboarding dropdown tableview that resizes based on content
+class OnboardingSelectTableView: UITableView {
+    let maxHeight: CGFloat = 230.0
+
+    override func reloadData() {
+        super.reloadData()
+        self.invalidateIntrinsicContentSize()
+        self.layoutIfNeeded()
+    }
+
+    override var intrinsicContentSize: CGSize {
+        let height = min(contentSize.height, maxHeight)
+        return CGSize(width: contentSize.width, height: height)
+    }
+}
+
+class OnboardingSearchDropdownView: UIView {
+
+    // MARK: - Private View Vars
+    private let searchBar = UISearchBar()
+    private let tableView = OnboardingSelectTableView()
+
+    // MARK: - Private Data Vars
+    private weak var delegate: OnboardingSearchViewDelegate?
+    private var placeholder: String!
+    private let reuseIdentifier = "OnboardingDropdownCell"
+    private var resultsTableData: [String] = []
+    private var tableData: [String]!
```

Comment:
Can these be non-optional if they're passed in through init?

Reply:
can make `delegate` `placeholder` and `tableData` non-optional and do 
```
self.delegate = delegate
self.placeholder = placeholder
self.tableData = tableData 
```
before `super.init` in the init function